blogposts/2023-02-10-adding-snapshot-testing-to-rnw-apps.md

```diff
@@ -0,0 +1,93 @@
+---
+post_title: "Adding Jest Snapshot Testing to React Native Windows Apps"
+author1: chiaramooney
+post_slug: 2023-02-10-adding-snapshot-testing-to-rnw-apps
+post_date: 2023-02-10 12:00:00
+categories: react-native
+tags: jest, react-native, testing, react-native-windows, tutorial
+summary: I added Jest Snapshot testing to a React Native Windows app with community modules. Here's a guide on how I did it!
+---
+
+[React Native Gallery](https://github.com/microsoft/react-native-gallery) is a React Native Windows app which showcases core and community module component samples. One function of the app is to validate whether a release has caused UI changes to our component set. In the past, this process was done manually by upgrading the app and walking through all sample pages to confirm that components were being rendered as expected, following the version upgrade. This process was time consuming and lacked accuracy. To automate more of this process, we've begun to add UI testing to React Native Gallery. In this document, I'll walk through how I added Jest Snapshot testing to React Native Gallery. 
+
+One of the most common testing frameworks to use with React Native apps is [Jest](https://jestjs.io/). Jest is a JavaScript testing framework, which is included by default within React Native apps that were created after v0.38. You can confirm that your app has Jest setup by checking the `package.json` file at the root of your app. It should contain the following code:
+```json
+{
+  "scripts": {
+    "test": "jest"
+  },
+  "jest": {
+    "preset": "react-native"
+  }
+}
+```
+
+[Snapshot tests](https://jestjs.io/docs/snapshot-testing) help to capture UI and verify that no unexpected UI changes have been made. A snapshot test case will include rendering a UI component, taking a snapshot, and then comparing the snapshot to a reference file. If the two do not match, the test will fail. This is will occur in one of two cases. Either the UI change is unexpected, or the change is expected, and the snapshot reference needs to be updated.
```

Comment:
```suggestion
[Snapshot tests](https://jestjs.io/docs/snapshot-testing) help to capture UI and verify that no unexpected UI changes have been made. A snapshot test case will include rendering a UI component, taking a snapshot, and then comparing the snapshot to a reference file. If the two do not match, the test will fail. This will occur in one of two cases: either the UI change is unexpected, or the change is expected - and the snapshot reference needs to be updated.
```